src/lib/comp/comp.c

```diff
@@ -595,7 +595,7 @@ ssize_t PUBLIC zck_write(zckCtx *zck, const char *src, const size_t src_size) {
                             "chunk");
                 else
                     zck_log(ZCK_LOG_DDEBUG, "Automatically ending chunk");
-                if(zck->comp.dc_data_size < zck->chunk_auto_min) {
+                if(!zck->no_check_min_size && zck->comp.dc_data_size < zck->chunk_auto_min) {
```

Comment:
I will retest and check again

Reply:
For reason I have not yet understood, the check is depending on the compression. Just disabling thwe check as in patch I get the results I am expecting. What I am doing:
- I run zck with -u option on a file (650MB)
- I have written a piece of code to call the library to generate the header from the same file, but without compression. That means I call zck_set_ioption(zck, ZCK_COMP_TYPE, ZCK_COMP_NONE) and zck_set_ioption(zck, ZCK_UNCOMP_HEADER, 1) before zck_write()
To be able to decide later which chunks should be installed, in both cases  I should get the same number of chunks (because I am using the same source file for testing) and chunks should have the same digest for uncompressed data.
This is what I get with my patch. However, if not applied, even if both are running comp_init() (from zck_write) and setting the same values for min and max in autochunk mode, zck (compressing with zst) reports a lot of "small chunks", and the two results are quite different.

I guessed I could have the same effect by zck_set_ioption(zck, ZCKCHUNK_MIN, 1). However, this runs in a chicken egg issue. It should be possible to set this before starting, that means before calling zck_write. This is not true because default value are set later by zck_write (in comp_init), and if I call it before zck_write I get:

comp_ioption: Minimum chunk size must be <= maximum chunk size
Minimum chunk size must be <= maximum chunk size

This is clear from code, but IMO the check should be done later (after zck_write is called) and not when setting during zck_set_option(). For example, I could require to set ZCKCHUNK_MIN but not ZCKCHUNK_MAX or viceversa.